forge-game/src/main/java/forge/game/cost/Cost.java

```diff
@@ -257,7 +257,7 @@ public Cost(String parse, final boolean bAbility) {
 
         }
 
-        if (parsedMana == null && (manaParts.length() > 0 || xCantBe0)) {
+        if (parsedMana == null && (manaParts.length() > 0 || manaParts.toString().contains("X") && xCantBe0)) {
```

Comment:
That's indeed ugly but currently will break Rasputin because the XCantBe0 will get lost
so you can activate his first ability even if he doesn't have counters

Reply:
it seems `XCantBe0$` might be supported instead

Reply:
I think that change should be kept